code/__HELPERS/icon_smoothing.dm

```diff
@@ -298,44 +374,82 @@ DEFINE_BITFIELD(smoothing_junction, list(
 
 	var/smooth_border = (smoothing_flags & SMOOTH_BORDER)
 	var/smooth_obj = (smoothing_flags & SMOOTH_OBJ)
-
-	#define SET_ADJ_IN_DIR(direction, direction_flag) \
-		set_adj_in_dir: { \
-			do { \
-				var/turf/neighbor = get_step(src, direction); \
-				if(neighbor) { \
-					var/neighbor_smoothing_groups = neighbor.smoothing_groups; \
-					if(neighbor_smoothing_groups) { \
-						for(var/target in canSmoothWith) { \
-							if(canSmoothWith[target] & neighbor_smoothing_groups[target]) { \
-								new_junction |= direction_flag; \
-								break set_adj_in_dir; \
-							}; \
+	var/border_object_smoothing = (smoothing_flags & SMOOTH_BORDER_OBJECT)
+
+	// Did you know you can pass defines into other defines? very handy, lets take advantage of it here to allow 0 cost variation
+	#define SEARCH_ADJ_IN_DIR(direction, direction_flag, ADJ_FOUND, WORLD_BORDER, BORDER_CHECK) \
```

Comment:
why are these defines nested inside of a proc? Is there some esoteric reason for it that I'm unaware of? It would make more sense for this to be moved out of the proc and de-indented (at least to reduce the ugliness of the indentation that this complex macro has) imo.

Reply:
also should we have check_ticks built in or is this quick enough?

Reply:
mostly because they're specific in scope to this proc, it's like a def undef at the start/end of the file but more particular (I realize I need to undef these now btw, will do that)
I can move them up if you'd like

Reply:
> also should we have check_ticks built in or is this quick enough?

You do not want check tick stuff in here, since this is part of a subsystem. it's also genuinely not all that expensive, so it's ok

Reply:
yes please move and undef thanks.

Reply:
actually I'm a bit uncomfortable now. The macros currently rely on being called in the scope of the proc, like, they rely implicitly on vars defined right above the macro definition. 

part of me is worried that moving it up will make that less clear, and fuck this cluster even further. Is there anything else I can do here to make the code more readable?

Reply:
if there's a reason for it to be unreadable it's fine, you hadn't mentioned the implicit vars

Reply:
I uh, forgot about em